File: python/copass-management/src/copass_management/tools/get_source.py
from __future__ import annotations

from typing import Any, Dict, TYPE_CHECKING

if TYPE_CHECKING:
    from copass_management.registrar import ToolContext


async def get_source(ctx: "ToolContext", input: Dict[str, Any]) -> Any:
    return await ctx.client.sources.retrieve(
        ctx.sandbox_id,
        str(input["data_source_id"]),
    )

File: typescript/packages/management/src/tools/get_source.ts
import type { ToolContext, ToolHandler } from '../registrar.js';

export const getSource: ToolHandler = async (
  ctx: ToolContext,
  input: Record<string, unknown>,
) => {
  const dataSourceId = String(input.data_source_id);
  return ctx.client.sources.retrieve(ctx.sandboxId, dataSourceId);
};
